bin/update-readmes.js

```diff
@@ -1,70 +1,138 @@
+
+/**
+ * Pattern matching start token of a README file.
+ *
+ * @type {RegExp}
+ */
+const TOKEN_PATTERN = /<!-- START TOKEN\((.+?(?:\|(.+?))?)\) -->/g;
+
+/**
+ * Given an absolute file path, returns the package name.
+ *
+ * @param {string} file Absolute path.
+ *
+ * @return {string} Package name.
+ */
+function getFilePackage( file ) {
+	return relative( PACKAGES_DIR, file ).split( sep )[ 0 ];
+}
+
+/**
+ * Returns an appropriate glob pattern for the packages directory to match
+ * relevant for a given set of files.
+ *
+ * @param {string[]} files Set of files to match. Pass an empty set to match
+ *                         all packages.
+ *
+ * @return {string} Packages glob pattern.
+ */
+function getPackagePattern( files ) {
+	if ( ! files.length ) {
+		return '*';
 	}
 
-	const [ packageName, targets ] = entry;
-
-	Object.entries( targets ).forEach( ( [ token, path ] ) => {
-		// Each target operates over the same file, so it needs to be processed synchronously,
-		// as to make sure the processes don't overwrite each other.
-		const { status, stderr } = spawnSync(
-			join( __dirname, '..', 'node_modules', '.bin', 'docgen' ).replace( / /g, '\\ ' ),
-			[
-				join( 'packages', packageName, path ),
-				`--output packages/${ packageName }/README.md`,
-				'--to-token',
-				`--use-token "${ token }"`,
-				'--ignore "/unstable|experimental/i"',
-			],
-			{ shell: true },
-		);
-
-		if ( status !== 0 ) {
-			process.stderr.write( `${ packageName } ${ stderr.toString() }\n` );
-			process.exit( 1 );
+	// Since brace expansion doesn't work with a single package, special-case
+	// the pattern for the singular match.
+	const packages = files.map( getFilePackage );
+	return packages.length === 1 ?
+		packages[ 0 ] :
+		'{' + packages.join() + '}';
+}
+
+/**
+ * Stream transform which filters out README files to include only those
+ * containing matched token pattern, yielding a tuple of the file and its
+ * matched tokens.
+ *
+ * @type {Transform}
+ */
+const filterTokenTransform = new Transform( {
+	objectMode: true,
+
+	async transform( file, _encoding, callback ) {
+		let content;
+		try {
+			content = await readFile( file, 'utf8' );
+		} catch {}
+
+		if ( content ) {
+			const tokens = [];
+
+			for ( const match of content.matchAll( TOKEN_PATTERN ) ) {
+				const [ , token, path = 'src/index.js' ] = match;
+				tokens.push( [ token, path ] );
+			}
+
+			if ( tokens.length ) {
+				this.push( [ file, tokens ] );
+			}
 		}
-	} );
+
+		callback();
+	},
 } );
+
+/**
+ * Optional process arguments for which to generate documentation.
+ *
+ * @type {string[]}
+ */
+const files = process.argv.slice( 2 );
+
+glob.stream( `${ PACKAGES_DIR }/${ getPackagePattern( files ) }/README.md` )
+	.pipe( filterTokenTransform )
+	.on( 'data', async ( /** @type {WPReadmeFileData} */ data ) => {
+		const [ file, tokens ] = data;
+
+		// Each target operates over the same file, so it needs to be processed
```

Comment:
This comment could be improved to something along these lines:

```
		// Each file can have more than one placeholder content to update
		// (which are represented by tokens). docgen updates one token at a time,
		// so we need to fill the tokens sequentially as to prevent that
		// the processes override each other.
```

Reply:
> This comment could be improved to something along these lines

Yep, I didn't think to update this, but I agree, terms like "target" don't make much sense anymore. Updated in 41908161d5f78dab12727566caaa2e5c1ebdc474.